infra/modules/cloudfront/main.tf

```diff
@@ -0,0 +1,207 @@
+terraform {
+  required_providers {
+    aws = {
+      source                = "hashicorp/aws"
+      configuration_aliases = [aws.us-east-1-provider]
+    }
+  }
+}
+
+data "aws_ssm_parameter" "api_gateway_endpoint" {
+  provider = aws
+  name     = "api_gateway_endpoint_${var.environment}_${var.project_name}"
+}
+
+data "aws_ssm_parameter" "api_certificate_arn" {
+  provider = aws
+  name     = "api_certificate_arn_${var.environment}_${var.project_name}"
+}
+
+data "aws_cloudfront_cache_policy" "caching_disabled" {
+  name = "Managed-CachingDisabled"
+}
+
+data "aws_cloudfront_origin_request_policy" "all_viewer_except_host_header" {
+  name = "Managed-AllViewerExceptHostHeader"
+}
+
+resource "aws_wafv2_web_acl" "api_web_acl" {
+  provider = aws.us-east-1-provider
+  name     = "${var.project_name}-${var.environment}-web-ACL"
+  scope    = "CLOUDFRONT"
+  default_action {
+    allow {
+    }
+  }
+  rule {
+    name     = "${var.project_name}-${var.environment}-RateBasedRule-IP-300"
+    priority = 0
+    action {
+      count {
+      }
+    }
+    statement {
+      rate_based_statement {
+        aggregate_key_type = "IP"
+        limit              = 300
+      }
+    }
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "${var.project_name}-${var.environment}-RateBasedRule-IP-300"
+      sampled_requests_enabled   = true
+    }
+  }
+  rule {
+    name     = "AWS-AWSManagedRulesAmazonIpReputationList"
+    priority = 1
+    override_action {
+      none {
+      }
+    }
+    statement {
+      managed_rule_group_statement {
+        name        = "AWSManagedRulesAmazonIpReputationList"
+        vendor_name = "AWS"
+        version     = null
+      }
+    }
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "AWS-AWSManagedRulesAmazonIpReputationList"
+      sampled_requests_enabled   = true
+    }
+  }
+  rule {
+    name     = "AWS-AWSManagedRulesCommonRuleSet"
+    priority = 2
+    override_action {
+      none {
+      }
+    }
+    statement {
+      managed_rule_group_statement {
+        name        = "AWSManagedRulesCommonRuleSet"
+        vendor_name = "AWS"
+        version     = null
+        rule_action_override {
```

Comment:
Is this the one that removes the size restrictions?

Reply:
How does it work? Just override it, don't set anything specific and it's disabled?

Reply:
Yep. The default action is to block, so this is overriding it with count, which means it will just "take note" and move on.

Reply:
Ah, maybe add a comment here explaining what this is doing. Either way I'll trust your wisdom here.